Internal: Create the basic e2e playwright test [ED-20640]

Ntnelbaba opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Add ID attributes to UI components to enable end-to-end testing with Playwright for the variables manager feature.
Main changes:
- Added unique ID attributes to color controls, font selectors, buttons, and popover components
- Created utility functions for variable management testing in a new playwright test file
- Implemented basic e2e tests for font and color variable creation and verification
- Added template file and updated npm scripts to support Playwright testing

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using. **[We'd love your feedback!](mailto:product@linearb.io)** 🚀</sub>
<!--end_gitstream_placeholder-->

Ntnelbaba committed:
Internal: Create the basic e2e playwright test [ED-20640]

Ntnelbaba committed:
Update lock in packages

Ntnelbaba committed:
Lint issues

Ntnelbaba committed:
Add single script for e2e

Ntnelbaba committed:
Move e2e to the repo root dir

Ntnelbaba committed:
lint

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Comments

Ntnelbaba committed:
Merge branch 'ED-20640-example-playwright-test-variables-manager' of github.com:elementor/elementor into ED-20640-example-playwright-test-variables-manager

Ntnelbaba commented on e2e/editor-variables-e2e/base.fixture.ts:
```diff
@@ -0,0 +1,15 @@
+import { setDefaultTemplate, setExperiment } from 'playwright';
```
This is a fixture-based file; it supports a default setup for a full suite
It will always be relevant for all the tests that need the variable manager test

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Svitlana-Dykun commented on playwright/pages/editor.ts:
```diff
@@ -0,0 +1,29 @@
+	return await extractPageId( page );
+};
+
+export const openNewPost = async ( page: Page ) => {
+	await page.goto( `/wp-admin/post-new.php` );
+	return await extractPageId( page );
+};
+
+const extractPageId = async ( page: Page ) => {
+	try {
+		await page.getByRole( 'button', { name: 'Close' } ).filter( { visible: true } ).click( { timeout: 1000 } );
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.log( 'Welcome to the editor message not found, skipping' );
+	}
+	await page.locator( '#elementor-switch-mode-button' ).or( page.locator( '#elementor-edit-mode-button' ) ).click();
+	await page.waitForSelector( editorPanelHeaderWrapperLocator, { state: 'visible', timeout: 10000 } );
+	await page.waitForTimeout( 1000 );
+	return page.url().split( 'post=' )[ 1 ].split( '&' )[ 0 ];
+};
```
Please use existing methods, see openNewPage tests/playwright/pages/wp-admin-page.ts

Svitlana-Dykun commented on playwright/playwright.config.ts:
Why do we need one more configuration? We can use existing one, the only thing that is changes is port, we can reuse it from .env for example

Ntnelbaba replied:
Changed to be base and reused here so it can be maintained in one location (this is temporary anyway)

Svitlana-Dykun commented on playwright/pages/library.ts:
```diff
@@ -0,0 +1,29 @@
+import { openNewPage } from './editor';
+import { canvasPageFrameLocator, pageLibraryLocator } from '../consts';
+
+export const importTemplate = async ( page: Page, templateName: string, templatePath: string ): Promise<string | undefined> => {
```
why not existing loadTemplate method?

Svitlana-Dykun commented on playwright/pages/experiments.ts:
```diff
@@ -0,0 +1,17 @@
+ * @param {string}                              experiment - Should be the exact e_NAME
+ * @param {('active' | 'inactive' | 'default')} state      - The state of the experiment
+ */
+export const setExperiment = async ( page: Page, experiment: string, state: 'active' | 'inactive' | 'default' = 'active' ) => {
```
setExperiments method exists on tests/playwright/pages/wp-admin-page.ts page. Why we can't use it?

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Change the main config to be base config

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
PR review

Ntnelbaba committed:
Merge branch 'ED-20640-example-playwright-test-variables-manager' of github.com:elementor/elementor into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Make test flow work

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Svitlana-Dykun commented on playwright/playwright.config.ts:
```diff
@@ -0,0 +1,21 @@
+import { resolve } from 'path';
```
please use existing config

Ntnelbaba committed:
Remove stuff

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Move stuff around

Ntnelbaba committed:
Merge branch 'ED-20640-example-playwright-test-variables-manager' of github.com:elementor/elementor into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Fix errors

Ntnelbaba committed:
Merge branch 'ED-20640-example-playwright-test-variables-manager' of github.com:elementor/elementor into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

elementorbot committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Update package.json

Ntnelbaba committed:
revert lock

Ntnelbaba committed:
Internal: Variables manager base tests [ED-20641] (#32589)

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:
https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no
spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Add UI element IDs to enable testing of the variables manager
functionality in the Elementor editor.
Main changes:
- Added unique ID attributes to UI components throughout the variables
manager interface
- Implemented utility functions for variables manager testing operations
- Created test cases for font and color variable creation and
verification
- Updated headless browser configuration for consistent test execution

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before
using. **[We'd love your feedback!](mailto:product@linearb.io)** 🚀</sub>
<!--end_gitstream_placeholder-->

Svitlana-Dykun committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Resolve the state storage file creation

Ntnelbaba committed:
add consts to selectors

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Resolve issue with test

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Use default new page create

Ntnelbaba committed:
Merge branch 'ED-20640-example-playwright-test-variables-manager' of github.com:elementor/elementor into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Add more experiments active to test

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Revert some changes

Ntnelbaba committed:
Merge branch 'main' into ED-20640-example-playwright-test-variables-manager

Ntnelbaba committed:
Revert more files

Ntnelbaba committed:
Revert package change

Ntnelbaba committed:
Move template to the inner folder

Ntnelbaba committed:
Skip new tests (experiments not working)

Ntnelbaba committed:
Empty commit
